feat: Automate token start date and integrate membership plans

SharolRI opened the pull request:
- Set the default start date to the current date in token generation mode
- Add a membership selection dropdown, replacing the manual end date
- Automatically calculate the end date based on the selected membership term_months
- Pass membership data to the location index view
- Update the generateToken method to handle membership_id and automatically calculate the end date
- Remove the manual end date from the token generation form
- Fix the 404 error on the membership page by placing the path in the web.php file

SharolRI committed:
feat: Automate token start date and integrate membership plans

- Set the default start date to the current date in token generation mode
- Add a membership selection dropdown, replacing the manual end date
- Automatically calculate the end date based on the selected membership term_months
- Pass membership data to the location index view
- Update the generateToken method to handle membership_id and automatically calculate the end date
- Remove the manual end date from the token generation form
- Fix the 404 error on the membership page by placing the path in the web.php file

And-Y21 commented on app/Http/Controllers/LocalityController.php:
```diff
@@ -31,8 +32,8 @@ public function index(Request $request)
-        
-        return view('localities.index', compact('localities','mailExamples'));
+
+        return view('localities.index', compact('localities','mailExamples', 'memberships')); // Agregar memberships aquí
```
Quitar comentarios.

And-Y21 commented on app/Http/Controllers/LocalityController.php:
```diff
@@ -21,6 +21,7 @@ public function index(Request $request)
         }
 
         $localities = $query->paginate(10);
+        $memberships = \App\Models\Membership::all();
```
Puedes mandar a llamar directamente un modelo en la parte de hasta arriba.

And-Y21 commented on app/Http/Controllers/LocalityController.php:
```diff
@@ -93,14 +94,17 @@ public function generateToken(Request $request)
-        $token = Token::generateTokenForLocality($id, $startDate, $endDate);
+        $membershipId = $request->input('membership_id');
+
+        $membership = \App\Models\Membership::find($membershipId);
```
Aqui tambien puedes ahorrar palabras mandando a llamar al modelo de Membresias.

SharolRI committed:
feat: Full membership integration for the token system

- Added membership model import in LocalityController
- Fixed missing request imports and removed duplicate code
- Followed Laravel conventions with direct model imports
- Removed a comment from app/Http/Controllers/LocalityController.php
